design/object-bucket/object-bucket.md

```diff
@@ -72,40 +62,36 @@ _As an admin, I want to quickly list all dynamically created buckets in a cluste
- 
+
 //TODO update diagram to reflect API changes  
- 
+
 ![user-actions](./cobc-user.png)
 
 1. The user creates an OBC in their namespace.
 1. The operator detects the new OBC instance.
-    1. The operator checks the `provisioner` field of the OBC's StorageClass.  If it matches, continue to the next step. Else, end the process here.
-    1. The operator gets the object store service using `storageClass.parameters.objectStoreNamespace` + `storageClass.parameters.objectStore`.
-1. The operator creates a CephObjectUser if it does not exist in order to generate the operator's access key.
-1. The operator creates the bucket using the service endpoint and access key.  The bucket name will be a concatenation of the  OBC's `metadata.namespace` and `metadata.name`.
+    1. The operator checks the `provisioner` field of the OBC's StorageClass.  If it matches, continue to the next step. Else, the process ends.
+    1. The object store service name is derived from the object
+1. The operator creates a new object store user.
+1. The operator creates the bucket using the object store endpoint and access key.
     1. If the bucket already exists in the store, the operator will instead regenerate the ConfigMap and Secret associated with that bucket.  This allows users the ability to reacquire buckets who's claims have been deleted.
-1. The operator creates an OB containing metadata about the new bucket and sets it as bound to the OBC.
-1. The operator updates the OBC to set it as "bound" to the OB.
+1. The operator creates an OB and sets it as bound to the OBC.
+1. The operator updates the OBC to reference the OB.
 1. The operator creates a ConfigMap in the namespace of the OBC with the endpoint information for that bucket.
-    1. The operator uses the radosgw-admin CLI to generate the user's access key.  
-    1. The key is linked to the bucket.
-1. The operator creates a Secret in the namespace of the OBC with the **user's** access key.
+1. The operator creates a Secret in the namespace of the OBC with the new user's key pair.
 1. An app Pod may then reference the Secret and the ConfigMap to begin accessing the bucket.
 
 **Use Case: Delete an Object Bucket**
 
 _As a Kubernetes user, I want to delete ObjectBucketClaim instances and cleanup generated API resources._
 
-1. The user deletes the OBC via `kubectl delete objectbucketclaim -n default FOO`.
-1. The OBC is marked for deletion and left in the foreground and the `Delete()` method, implemented by this provisioner, is invoked by the bucket lib.
-1. The respective ConfigMap and user Secret are deleted (done by the bucket lib).
-1. The OBC is garbage collected.
-1. The OB phase is changed from "bound" to "lost"
-
-    Note: this does not cause the deletion of the OB.  Since the actual bucket is not deleted, it is reasonable that it should still tracked in Kubernetes.
+1. The user deletes the OBC
```

Comment:
This describes k8s operations performed by the lib but doesn't mention the cleanup done by the ceph provisioner.